### PR TITLE
fix(cas): Check if timed out transaction succeeded when receiving a nonce error

### PR DESCRIPTION
**Pull request by stbrody**
builds on https://github.com/ceramicnetwork/ceramic-anchor-service/pull/335

### CODE REVIEW

File: src/services/blockchain/__tests__/eth-bc-service.test.ts [hunk 1]
```diff
@@ -237,38 +247,91 @@ describe('ETH service with mock wallet',  () => {
...
+    const [txData2, attemptNum2, network2] = mockTrySendTransaction.mock.calls[2]
     expect(attemptNum2).toEqual(2)
     expect(network2).toEqual(config.blockchain.connectors.ethereum.network)
-    expect(transactionTimeoutSecs2).toEqual(config.blockchain.connectors.ethereum.transactionTimeoutSecs)
     expect(txData2).toMatchSnapshot()
+
+    expect(mockConfirmTransactionSuccess).toHaveBeenCalledTimes(3)
+
+    const [txResponse0, network3, transactionTimeoutSecs0] = mockConfirmTransactionSuccess.mock.calls[0]
+    expect(network3).toEqual(config.blockchain.connectors.ethereum.network)
```
Comment (valmack) [inline src/services/blockchain/__tests__/eth-bc-service.test.ts:287]: Why do we need to check the network in every result for this test?

**Review by valmack — COMMENTED** [review]
Nice improvements. Just a few comments

File: src/services/blockchain/__tests__/eth-bc-service.test.ts (hunk 1 above)
Comment (stbrody) [reply src/services/blockchain/__tests__/eth-bc-service.test.ts]: Yeah, I guess I was just being thorough but you're right it's probably overkill and making the test unnecessarily brittle.  Updated

File: src/services/blockchain/__tests__/eth-bc-service.test.ts [hunk 2]
```diff
@@ -237,38 +247,91 @@ describe('ETH service with mock wallet',  () => {
...
+    provider.estimateGas.mockReturnValue(gasEstimate)
+
+    const mockTrySendTransaction = jest.fn()
+    const mockConfirmTransactionSuccess = jest.fn()
+    ethBc._trySendTransaction = mockTrySendTransaction
+    ethBc._confirmTransactionSuccess = mockConfirmTransactionSuccess
+    // first time submits the txn successfully
+    mockTrySendTransaction.mockReturnValueOnce(txResponse)
+    // On retry we get a NONCE_EXPIRED error because the first was actually mined correctly
+    mockTrySendTransaction.mockRejectedValueOnce({code: ErrorCode.NONCE_EXPIRED})
```
[the remark this answers is not in the document]
Comment (stbrody) [reply src/services/blockchain/__tests__/eth-bc-service.test.ts:293]: What do you mean?  Thrown by what?  The mock?  Doesn't seem useful to test the behavior of jest itself.

[the remark this answers is not in the document]
Comment (valmack) [reply src/services/blockchain/__tests__/eth-bc-service.test.ts:293]: I meant test that an error with code `ErrorCode.NONCE_EXPIRED` actually gets thrown

[the remark this answers is not in the document]
Comment (stbrody) [reply src/services/blockchain/__tests__/eth-bc-service.test.ts:293]: Right, isn't that basically just testing that the mock did what it was supposed to?  We know it was called twice from the check below, and we know from this line and the one above it that the second time the mock is called it's supposed to throw NONCE_EXPIRED.  I'm actually not sure how I would even go about testing that here to be honest.

[the remark this answers is not in the document]
Comment (valmack) [reply src/services/blockchain/__tests__/eth-bc-service.test.ts:293]: Ok I understand now. We return resp, then get a timeout, then get an expired nonce, then a success from first nonce. 😅 wasn't immediately obvious what we are testing

**Review by valmack — APPROVED** [review]
👍

**Pull request merged**